### PR TITLE
Pfc tests migration from abstract to snappi

### PR DESCRIPTION
**Pull request by ANISH-GOTTAPU**
### Description of PR
Migration of PFC tests from `ixnetwork-open-traffic-generator` to `snappi`

Summary:
Test case(new/improvement)

### Type of change
- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x ] Test case(new/improvement)

#### What is the motivation for this PR?
The motivation for this PR is to migrate the tests that are written with `ixnetwork open traffic generator` to `snappi`

#### How did you do it?
Replaced the code where abstract is used to snappi

#### How did you verify/test it?
verified all the tests in sonic-mgmt using pytest with DUT and IXIA connections 
snappi version used for the development of tests `snappi[ixnetwork]==0.4.1`

### CODE REVIEW

File: tests/common/fixtures/snappi_fixtures.py [hunk 1]
```diff
@@ -0,0 +1,425 @@
+"""
```
Comment (baiwei0427) [inline tests/common/fixtures/snappi_fixtures.py:1]: This file is also created in https://github.com/Azure/sonic-mgmt/pull/3653/

File: tests/snappi/conftest.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
+import pytest
+import random
+from tests.common.ixia.common_helpers import enable_packet_aging, start_pfcwd
```
Comment (baiwei0427) [inline tests/snappi/conftest.py:3]: Use tests.common.snappi.common_helpers instead.

File: tests/snappi/pfc/test_global_pause_with_snappi.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
+import pytest
+
+from tests.common.helpers.assertions import pytest_require
+from tests.common.fixtures.conn_graph_facts import conn_graph_facts,\
+    fanout_graph_facts
+from tests.common.fixtures.snappi_fixtures import snappi_api_serv_ip, snappi_api_serv_port,\
+    snappi_api, tgen_testbed_config
+from tests.common.ixia.qos_fixtures import prio_dscp_map, all_prio_list, lossless_prio_list,\
```
Comment (baiwei0427) [inline tests/snappi/pfc/test_global_pause_with_snappi.py:8]: ixia -> snappi

File: tests/snappi/pfc/test_global_pause_with_snappi.py [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    lossy_prio_list
+
+from files.helper import run_pfc_test
+
+pytestmark = [pytest.mark.topology('tgen')]
```
Comment (baiwei0427) [inline tests/snappi/pfc/test_global_pause_with_snappi.py:13]: @pytest.mark.topology("snappi")

File: tests/snappi/pfc/test_pfc_pause_lossless_with_snappi.py [hunk 5]
```diff
@@ -0,0 +1,264 @@
+import logging
+import pytest
+
+from files.helper import run_pfc_test
+from tests.common.helpers.assertions import pytest_assert, pytest_require
+from tests.common.fixtures.conn_graph_facts import conn_graph_facts,\
+    fanout_graph_facts
+from tests.common.fixtures.snappi_fixtures import snappi_api_serv_ip, snappi_api_serv_port,\
+    snappi_api, tgen_testbed_config
+from tests.common.ixia.qos_fixtures import prio_dscp_map, all_prio_list, lossless_prio_list,\
```
Comment (baiwei0427) [inline tests/snappi/pfc/test_pfc_pause_lossless_with_snappi.py:10]: ixia -> snappi

File: tests/snappi/pfc/test_pfc_pause_lossless_with_snappi.py [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+    lossy_prio_list
+from tests.common.reboot import reboot
+from tests.common.utilities import wait_until
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [pytest.mark.topology('tgen')]
```
Comment (baiwei0427) [inline tests/snappi/pfc/test_pfc_pause_lossless_with_snappi.py:17]: tgen -> snappi

File: tests/snappi/pfc/test_pfc_pause_lossy_with_snappi.py [hunk 7]
```diff
@@ -0,0 +1,262 @@
+import logging
+import pytest
+
+from files.helper import run_pfc_test
+from tests.common.helpers.assertions import pytest_assert, pytest_require
+from tests.common.fixtures.conn_graph_facts import conn_graph_facts,\
+    fanout_graph_facts
+from tests.common.fixtures.snappi_fixtures import snappi_api_serv_ip, snappi_api_serv_port,\
+    snappi_api, tgen_testbed_config
+from tests.common.ixia.qos_fixtures import prio_dscp_map, all_prio_list, lossless_prio_list,\
```
Comment (baiwei0427) [inline tests/snappi/pfc/test_pfc_pause_lossy_with_snappi.py:10]: ixia -> snappi

File: tests/snappi/pfc/test_pfc_pause_lossy_with_snappi.py [hunk 8]
```diff
@@ -0,0 +1,262 @@
...
+    lossy_prio_list
+from tests.common.reboot import reboot
+from tests.common.utilities import wait_until
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [pytest.mark.topology('tgen')]
```
Comment (baiwei0427) [inline tests/snappi/pfc/test_pfc_pause_lossy_with_snappi.py:17]: tgen -> snappi

**Review by baiwei0427 — CHANGES_REQUESTED** [review]

**Comment by ANISH-GOTTAPU** [comment]
@baiwei0427 : This PR was raised before test_snappi, once test_snappi is merged, I will pull the master and update this PR accordingly

**Comment by baiwei0427** [comment]
> @baiwei0427 : This PR was raised before test_snappi, once test_snappi is merged, I will pull the master and update this PR accordingly

Thanks! Please change all 'ixia' and 'tgen' to 'snappi'

**Comment by ANISH-GOTTAPU** [comment]
closing this as it is addressed in #3794

**Pull request closed without merging**